es/index.js

```diff
@@ -86,22 +100,11 @@ export function exponentialDelay(retryNumber = 0) {
  * @return {Object}
  */
 function getCurrentState(config) {
-  const currentState = config[namespace] || {};
+  const currentState = config.adapter[namespace] || {};
   currentState.retryCount = currentState.retryCount || 0;
-  config[namespace] = currentState;
```

Comment:
making this function pure(r) now that we don't actually need to store anything in config.

Reply:
We prefer not to use adapter to pass our retry config. Have a look at #82